sdk/resourcemanager/Azure.ResourceManager.Core/src/GenericResourceOperations.cs

```diff
@@ -30,11 +30,19 @@ internal GenericResourceOperations(ResourceOperationsBase operations, ResourceId
         /// <inheritdoc/>
         protected override ResourceType ValidResourceType => ResourceGroupOperations.ResourceType;
 
-        private ResourcesOperations Operations => new ResourcesManagementClient(
-            BaseUri,
-            Id.Subscription,
-            Credential,
-            ClientOptions.Convert<ResourcesManagementClientOptions>()).Resources;
+        private ResourcesOperations Operations
+        {
+            get
+            {
+                string subscription;
+                Id.TryGetSubscriptionId(out subscription);
+                return new ResourcesManagementClient(
+                    BaseUri,
+                    subscription,
```

Comment:
So, whatever we do here will change once we move to using the RestClients - subscriptionId will not be required to construct them.  I will add in a backoff to using Guid.Empty, but this will have to be removed when we move away from wrapping the management client.

Reply:
Updated for correctness.  However, I don't think null can occur here, because this class can only be used for tracked resources.